kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js

```diff
@@ -387,103 +390,114 @@ define([
 
-                return div({
-                    id: elementId,
-                    dataParameter: parameterId,
-                    dataAdvancedParameter: advanced
-                });
-            }).join('\n');
+                    return div({
+                        id: elementId,
+                        dataParameter: parameterId,
+                        dataAdvancedParameter: advanced,
+                    });
+                })
+                .join('\n');
 
             return {
                 content: layout,
                 layout: orderedParams,
                 params: params,
                 view: view,
-                paramMap: paramMap
+                paramMap: paramMap,
             };
         }
 
-        // LIFECYCLE API
-        function renderParameters() {
-            // First get the app specs, which is stashed in the model,
-            // with the parameters returned.
-            // Separate out the params into the primary groups.
-            const appSpec = model.getItem('appSpec');
-
-            return Promise.try(function () {
-                const params = model.getItem('parameters');
-                let parameterParams = makeParamsLayout(
-                    params.layout.filter(function (id) {
-                        const original = params.specs[id].original;
-
-                        let isParameter = false;
+        function createParameterWidget(appSpec, parameterInfo, parameterId) {
+            const spec = parameterInfo.paramMap[parameterId];
+            return makeFieldWidget(appSpec, spec, initialParams[spec.id])
+                .then((widget) => {
+                    widgets.push(widget);
+                    return widget.start({
+                        node: document.getElementById(parameterInfo.view[spec.id].id),
+                    });
+                })
+                .catch((ex) => {
+                    const errorDisplay = div(
+                        {
+                            class: 'kb-field-widget__error_message--parameters',
+                        },
+                        [ex.message]
+                    );
+                    document.getElementById(
+                        parameterInfo.view[spec.id].id
+                    ).innerHTML = errorDisplay;
 
-                        if (original) {
+                    throw new Error('Error making input field widget', ex);
+                });
+        }
 
-                            //looking for file inputs via the dynamic_dropdown data source
-                            if (original.dynamic_dropdown_options) {
-                                isParameter = original.dynamic_dropdown_options.data_source !== 'ftp_staging';
-                            }
+        /* 
+            Filter out any inputs which may not be parameters (e.g. file inputs, output fields), map these to the parameter ID so we can use it to create the expected layout
+        */
+        function filterParameters(params) {
```

Comment:
this was also moved out of that monster renderParameters() method. no refactoring of the code itself other than to move it into its own function